frontend/lib/src/features/account_details/account_details_page.dart

```diff
@@ -228,11 +226,23 @@ class AccountDetailsPage extends StatelessWidget {
               ),
               Padding(
                 padding: const EdgeInsets.fromLTRB(25, 0, 25, 20),
-                child: Graph(
-                  onTap: () {
-                    dashboardAppNav.currentState?.pushNamed(
-                      '/spendingBreakdown',
-                      arguments: ScreenArguments(arguments.accountId),
+                child: FutureBuilder(
+                  future: spendingBreakdownController.getSpendingBreakdown(
+                    accountId: arguments.accountId,
+                  ),
+                  builder: (context, snapshot) {
+                    if (snapshot.connectionState != ConnectionState.done) {
+                      return const CircularProgressIndicator();
+                    }
```

Comment:
it seems after your implementation this future builder is not needed anymore

graph.dart already calls the api
![image](https://github.com/framgia/sph-flutter/assets/95029803/fa4dd7e1-1435-490c-a066-25faf4e68d24)